pkg/resource/stack/deployment.go

```diff
@@ -870,6 +870,44 @@ func DeserializePropertyValue(v interface{}, dec config.Decrypter,
 
 			// Otherwise, it's just a weakly typed object map.
 			return resource.NewProperty(obj), nil
+		case *apitype.SecretV1:
+			prop := resource.MakeSecret(resource.NewNullProperty())
```

Comment:
Can we not share the code with deserialising from maps? Just make it an inner function of this method and translate SecretV1 into a map to call it